### DISCUSSION TITLE
where are the config files created in debian bookworm?

### DISCUSSION THREAD

**Discussion by NossieUK** [christgau/wsdd#171] [Q&A]
I have wsdd installed via apt - I feel really silly with this but can someone tell me

A: Where do I add the SMB_GROUP line to use the samba workgroup?
B: What do I edit to allow WSDD to use those flags on startup?

set the Workgroup according to smb.conf and be verbose

SMB_GROUP=$(grep -i '^\sworkgroup\s=' smb.conf | cut -f2 -d= | tr -d '[:blank:]')

wsdd -v -w $SMB_GROUP

just to add this is debian bookworm - there appears to be no clear way as to where the conf files are created?

**Comment by christgau** [comment]
> A: Where do I add the SMB_GROUP line to use the samba workgroup?

For systemd?! Nowhere, I assume(see below).

> B: What do I edit to allow WSDD to use those flags on startup?

Take a look at the place where most configuration files are located, your `/etc` directory. I haven't looked into bookworm, but I suspect there is a `wsdd.conf` file somewhere that may look like the [one in the repo](https://github.com/christgau/wsdd/blob/0aef2d73c01162d746414c25f1556ae510289cf4/etc/systemd/wsdd.defaults). 

However, AFAIK, you cannot execute shell commands in such a file. Thus, you have to specify the workgroup manually there. You can use the cited command line from the man page to determine the workgroup or just take a look into the samba configuration.

**Comment by simonmcnair** [comment]
I think it's /etc/default/wsdd

**Comment by simonmcnair** [comment]
sorry, my bad.  It's /usr/lib/systemd/system/wsdd.service

**Comment by simonmcnair** [comment]
oh.  /usr/lib/systemd/system/wsdd.service contains a reference to /etc/default/wsdd so I was correct.
